### DISCUSSION TITLE
Anyone interested  in a plain TCP transport implementation of JsSIP.Socket?

### DISCUSSION THREAD

**Discussion by pjkundert** [versatica/JsSIP#859] [Q&A] [answered]
I'd like the option of connecting directly with some SiP systems that only support UDP and TCP connections.  

Apparently, the JsSIP.Socket allows the creation of new transports not based on WebSockets, but I'm not sufficiently capable to figure out exactly ... how to do that.

Does anyone know of anyone else implementing alternative JsSIP.Socket transports, such as SIP via TCP/IP (with or without TLS)?

**Comment by ibc** [comment, marked as answer]
Do you know that you cannot open raw TCP or UDP connections from a browser app?

**Reply by pjkundert** [reply]
Ah, there you go!  Thanks,

**Reply by pjkundert** [reply]
I've implemented a full Asterisk PBX which accepts incoming SIP via  WebRTC, and routes traffic to the target SIP devices via UDP, so I've got a solution that works.  I guess, until the target devices themselves begin accepting WebRTC, that's what I'll have to be satisfied with.

**Comment by pjkundert** [comment]
For example, someone wrote a SIP.js TCP transport; it feels like a JsSIP.Socket TCP transport implementation would look something like this:  https://gist.github.com/phanmn/de0929fc4945c435cebfb6635366e87c

And, when @jmillan committed the changes to JsSIP.Socket, he tantalizingly claimed that it should allow the creation of non-WebSocket transports:  https://github.com/versatica/JsSIP/pull/374

**Comment by ROBERT-MCDOWELL** [comment]
webtransport (http3) seems intersting to explore for this topic
